### ISSUE TITLE
[HOLD for regression test] [$1000] Tooltip not shown for emoji picker button on hover the outer part, it only shown for hovering the icon

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#13455]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed: 
Hover on the emojipicker button icon surrounding pressable area

## Expected Result: 
Tooltip should shown for entire button

## Actual Result: 
Tooltip only shows on hover the icon

## Workaround:
unknown

## Platform:

Where is this issue occurring?

- Web

**Version Number:** 1.2.37-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
[ff5be293-b6dd-4ede-8cde-239f39379a3f.webm](https://user-images.githubusercontent.com/43996225/206594143-1247928b-1832-4e64-b9c6-abe041a40fd2.webm)

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @Pujan92 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1670532553163569

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

https://user-images.githubusercontent.com/43996225/206594156-1c8b2031-d564-43b2-ac46-d88917dbd0e3.mp4

[ff5be293-b6dd-4ede-8cde-239f39379a3f.webm](https://user-images.githubusercontent.com/43996225/206594204-fb5618d0-8b3e-472d-accb-9043db1a137a.webm)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e647d4e751c0a2f6</li>
        <li>Upwork Job ID: 1601033782775242752
<li>Last Price Increase: 2022-12-12</li></li>
    </ul>
</details>

**Comment by syedsaroshfarrukhdot** [comment]
**Proposal** :- 

We need to wrap `Pressable` in `Tooltip` instead of the `icon` 

```diff
diff --git a/src/components/EmojiPicker/EmojiPickerButton.js b/src/components/EmojiPicker/EmojiPickerButton.js
index 32af3ed59..f199850be 100644
--- a/src/components/EmojiPicker/EmojiPickerButton.js
+++ b/src/components/EmojiPicker/EmojiPickerButton.js
@@ -28,6 +28,7 @@ const defaultProps = {
 const EmojiPickerButton = (props) => {
     let emojiPopoverAnchor = null;
     return (
+        <Tooltip text={props.translate('reportActionCompose.emoji')}>
         <Pressable
             ref={el => emojiPopoverAnchor = el}
             style={({hovered, pressed}) => ([
@@ -39,14 +40,15 @@ const EmojiPickerButton = (props) => {
             nativeID={props.nativeID}
         >
             {({hovered, pressed}) => (
-                <Tooltip text={props.translate('reportActionCompose.emoji')}>
+               
                     <Icon
                         src={Expensicons.Emoji}
                         fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed))}
                     />
-                </Tooltip>
+               
             )}
         </Pressable>
+        </Tooltip>
     );
 };
```

**After Fix :-** 

https://user-images.githubusercontent.com/81307212/206596002-448e07f9-1491-4642-97a3-320f9e0d103f.mov

**Comment by Pujan92** [comment]
**Proposal**

Wrap `Pressable` inside the `Tooltip`.

https://github.com/Expensify/App/blob/216bbe9ff3995264e08a542d618bcc1db12d946f/src/components/EmojiPicker/EmojiPickerButton.js#L31-L49

```
     <Tooltip text={props.translate('reportActionCompose.emoji')}>
            <Pressable
                ref={el => emojiPopoverAnchor = el}
                style={({hovered, pressed}) => ([
                    styles.chatItemEmojiButton,
                    StyleUtils.getButtonBackgroundColorStyle(getButtonState(hovered, pressed)),
                ])}
                disabled={props.isDisabled}
                onPress={() => EmojiPickerAction.showEmojiPicker(props.onModalHide, props.onEmojiSelected, emojiPopoverAnchor)}
                nativeID={props.nativeID}
            >
                {({hovered, pressed}) => (
                    <Icon
                        src={Expensicons.Emoji}
                        fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed))}
                    />
                )}
            </Pressable>
        </Tooltip>
```

**Comment by kryptify** [comment]
Proposal:
You need to wrap whole pressable component with Tooltip. At this time, Tooltip is wrapping a Icon component.
That's why the tooltip is showing when mouse hover the Icon, not whole pressable.
Code is same as above.
My Upwork profile is :
https://www.upwork.com/freelancers/~01a1a69d7da842b796

**Comment by MunibQazi12** [comment]
**Proposal**
It is a simple problem to fix we need to just wrap the tooltip component into the pressable component.
This is a simple solution to this problem.
Here is my Upwork profile: https://www.upwork.com/freelancers/~01cf836885e7c7a016

**Comment by s77rt** [comment]
IMO @Pujan92 should be hired, it's the first (and probably best) working solution after `Help Wanted` and `External` labels are applied.

**Comment by syedsaroshfarrukhdot** [comment]
@s77rt Just to be clear there is no such rule that you can't propose solution before `External` label according to Contributing Guidelines. We are allowed to post proposals before `External` Label.

![Screenshot 2022-12-09 at 9 20 56 AM](https://user-images.githubusercontent.com/81307212/206679713-ccf55f05-ab77-4575-ba81-e3b0a55102bd.png)

As @Pujan92 is duplicate proposal to mine. I don't think they should be hired.

This concern was also raised in another issue https://github.com/Expensify/App/issues/13435#issuecomment-1343468927 they were advised to bring there concerns in slack channel by the assignee. I would also advise you to do the same for any concerns.

**Comment by Pujan92** [comment]
> @s77rt Just to be clear there is no such rule that you can't propose solution before `External` label according to Contributing Guidelines. We are allowed to post proposals before `External` Label.

I think one of the latest comment by @JmillsExpensify suggested locking the issue till the external label won't be added which indirectly point to the same thought according to me. Let the team decide and I am fine with their decision.

**Comment by syedsaroshfarrukhdot** [comment]
> Let the team decide and I am fine with their decision.

Yes, it would be the best let the team decide I will also be okay with there decision.

**Comment by asimimamabdullah** [comment]
Hi, I am struggling to set up this project locally. I need help, I am not here to steal anyone's work
This is the error I am getting while just running 'npm install'. I am using the same node version, can you help?


### **npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm ERR! code 1
npm ERR! path C:\Users\asimi\OneDrive\Desktop\work-practice\App\node_modules\shellcheck
npm ERR! command failed
**npm ERR! command C:\Windows\system32\cmd.exe /d /s /c ./install.sh**     
npm ERR! '.' is not recognized as an internal or external command,     
npm ERR! operable program or batch file.** ###

**Comment by thienlnam** [comment]
@asimimamabdullah Please direct all setup and environment problems not related to the issue to #expensify-open-source [here](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#slack-channels)

We're aware of the lack of clarity around proposals before the External label, and have a couple fixes that will be coming out soon. But yes, contributors should not be posting until the external label is applied.  I'm going to minimize all conversation that is not related to the issue.  If you have additional questions, please bring it to [#expensify-open-source](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#slack-channels)

**Comment by Pujan92** [comment]
1 minor issue I am seeing for the above proposals is due to the vertical margin applied to the emoji button showing tooltip even though we hover on the top and bottom part. 

**Issue**
[4734cc5f-15ea-45aa-af1f-d70ed50a66f8.webm](https://user-images.githubusercontent.com/14358475/206822573-71966dcb-054e-48de-bf2b-6c4b4edf4033.webm)

**Updated Proposal**

Wrap `Tooltip` with `View` and remove vertical margin from this emoji button.
```
<View style={[styles.justifyContentCenter]}>
    <Tooltip text={props.translate('reportActionCompose.emoji')}>
```

https://github.com/Expensify/App/blob/ed3e2dc507e0f02a30b22a012412fbecc0be2290/src/styles/styles.js#L1482
remove `marginVertical` property

[9c0cb7b4-9cbf-4adc-b389-ef96cd1ebf05.webm](https://user-images.githubusercontent.com/14358475/206822833-e044556d-3771-4ec2-a55c-0080b8f383f4.webm)

**Comment by Pujan92** [comment]
I don't think it is intentional and according to me, it might not be a good user experience. I have raised the same [here](https://github.com/Expensify/App/issues/13340). To make the item center I think it's better to use `justifyContent` instead of a fixed vertical margin.

**Comment by kadiealexander** [comment]
@aimane-chnaif any thoughts on the above?

**Comment by thienlnam** [comment]
IMO, we should keep it consistent with the send button and so when you hover between them it has the same height
so `4px margin between emoji button and tooltip (similar to send button)`

**Comment by situchan** [comment]
### **Proposal:**

Previous proposals didn't handle responsive with multiline text

https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPickerButton.js
```diff
    return (
+     <Tooltip containerStyles={styles.alignSelfEnd} text={props.translate('reportActionCompose.emoji')}>
        <Pressable
            ref={el => emojiPopoverAnchor = el}
            style={({hovered, pressed}) => ([
                styles.chatItemEmojiButton,
                StyleUtils.getButtonBackgroundColorStyle(getButtonState(hovered, pressed)),
            ])}
            disabled={props.isDisabled}
            onPress={() => EmojiPickerAction.showEmojiPicker(props.onModalHide, props.onEmojiSelected, emojiPopoverAnchor)}
            nativeID={props.nativeID}
        >
            {({hovered, pressed}) => (
-               <Tooltip text={props.translate('reportActionCompose.emoji')}>
                    <Icon
                        src={Expensicons.Emoji}
                        fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed))}
                    />
-               </Tooltip>
            )}
        </Pressable>
+     </Tooltip>
    );
```

**Comment by shawnborton** [comment]
Keeping it consistent with the send button makes sense to me, but ultimately I would think this one would also behave like other buttons, for example the mini menu:
<img width="167" alt="image" src="https://user-images.githubusercontent.com/2319350/207421494-1d144610-bcd3-460a-b973-e9e2eeb1bbfd.png">

**Comment by syedsaroshfarrukhdot** [comment]
Updated Proposal :- 

To handle edge case of multiline text in the composer we can wrap `EmojiPickerButton` with a `view` just like we are doing for send button as passing self align property to tooltip container doesn't sound a good idea.

```diff

diff --git a/src/components/EmojiPicker/EmojiPickerButton.js b/src/components/EmojiPicker/EmojiPickerButton.js
index 32af3ed59..237af6c43 100644
--- a/src/components/EmojiPicker/EmojiPickerButton.js
+++ b/src/components/EmojiPicker/EmojiPickerButton.js
@@ -28,6 +28,7 @@ const defaultProps = {
 const EmojiPickerButton = (props) => {
     let emojiPopoverAnchor = null;
     return (
+        <Tooltip text={props.translate('reportActionCompose.emoji')}>
         <Pressable
             ref={el => emojiPopoverAnchor = el}
             style={({hovered, pressed}) => ([
@@ -39,14 +40,13 @@ const EmojiPickerButton = (props) => {
             nativeID={props.nativeID}
         >
             {({hovered, pressed}) => (
-                <Tooltip text={props.translate('reportActionCompose.emoji')}>
                     <Icon
                         src={Expensicons.Emoji}
                         fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed))}
                     />
-                </Tooltip>
             )}
         </Pressable>
+        </Tooltip>
     );
 };
 
diff --git a/src/pages/home/report/ReportActionCompose.js b/src/pages/home/report/ReportActionCompose.js
index 7811d1c25..ad94fd561 100644
--- a/src/pages/home/report/ReportActionCompose.js
+++ b/src/pages/home/report/ReportActionCompose.js
@@ -686,11 +686,13 @@ class ReportActionCompose extends React.Component {
                         )}
                     </AttachmentModal>
                     {canUseTouchScreen() && this.props.isMediumScreenWidth ? null : (
+                        <View style={[styles.justifyContentEnd]}>
                         <EmojiPickerButton
                             isDisabled={isBlockedFromConcierge}
                             onModalHide={() => this.focus(true)}
                             onEmojiSelected={this.addEmojiToTextBox}
                         />
+                        </View>
                     )}
                     <View style={[styles.justifyContentEnd]}>
                         <Tooltip text={this.props.translate('common.send')}>

```


https://user-images.githubusercontent.com/81307212/207423409-a58747db-77d0-4237-8424-3dd5f456e719.mov

**Comment by syedsaroshfarrukhdot** [comment]
>  can you elaborate more detail about the reason?

@aimane-chnaif My bad at first glance I thought it might mess up the tooltip to show when setting tooltip container to `flex-end` so proposed the change https://github.com/Expensify/App/issues/13455#issuecomment-134953956 . On testing it also works pretty good and fixes the issue on multiline comment. Now, it is up to you to go with whose ever proposal. Both proposals do the same job and fix the issue. Only mine follows the same approach we are using to keep send button icon to `flex-end`

**Comment by thienlnam** [comment]
Removing the Help Wanted label since we have plenty of proposals to choose from that will solve the issue

**Comment by thienlnam** [comment]
Thanks for the review, I agree with your analysis.

In this specific scenario, it might not matter too much since the text is constant and unlikely to break into multiple lines but I'll move forward with @situchan's proposal since since they considered other edge-cases here and like you mentioned found a fix to a possible regression

**Comment by kadiealexander** [comment]
Thanks @thienlnam!! I've sent upwork contracts to @aimane-chnaif and @Pujan92.

@situchan please apply on [Upwork](https://www.upwork.com/jobs/~01e647d4e751c0a2f6) or let me know your name in Upwork so we can ensure you're paid :)

**Comment by kadiealexander** [comment]
I'm going OOO for a family emergency, if this needs action while I'm away please re-add the bug label to assign someone.

**Comment by situchan** [comment]
> @situchan please apply on Upwork or let me know your name in Upwork so we can ensure you're paid :)

@kadiealexander applied. thank you

**Comment by JmillsExpensify** [comment]
I extended the contract while Kadie is out. In the meantime, we're still waiting on the regression period to pass.

**Comment by JmillsExpensify** [comment]
I'm not assigned here but I can help. Assigning myself.

**Comment by JmillsExpensify** [comment]
It looks like we have the following:

 - Issue reporter: @Pujan92 $250
 - Contributor: @situchan $1,500 (incl. 50% bonus)
 - Contributor+: @aimane-chnaif $1,500 (incl. 50% bonus). 

Upwork job is here. All contributors have been paid out.

**Comment by JmillsExpensify** [comment]
That means we only lack the BZ checklist. @thienlnam do you agree with @aimane-chnaif that we're good on your items? If so, let's get those checked off.

**Comment by thienlnam** [comment]
Yeah I agree, checked those items off

**Comment by kadiealexander** [comment]
Thank you for your help @JmillsExpensify, I was unfortunately OOO sick. I've proposed a regression test update in Slack [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1673912860873499) :)

**Comment by JmillsExpensify** [comment]
Awesome thank you!

**Comment by thienlnam** [comment]
Closing this out, from the discussion here there is no regression test needed https://expensify.slack.com/archives/C01SKUP7QR0/p1673912860873499
